### PR TITLE
Improve the doc interface

### PR DESCRIPTION
**Pull request by aadcg**
This is actual trivial, since it suffices to add some keybindings.

But I'm having a hard time doing it since we need to rethink the modules of the Nyxt system.

Not sure how to proceed yet. Trying to slowly make sense of everything.

Ideas welcome.

Addresses #1541.

### CODE REVIEW

**Comment by Ambrevar** [comment]
Can you elaborate on what's blocking you?  Why do we need to rethinkg the Nyxt module system?
(I assume you mean the ASDF modules.)

**Comment by aadcg** [comment]
@Ambrevar, yes I was referring to the ASDF modules. I think that `help` should inherit from `web-mode`. Commands such as `jump-to-heading` are paramount for `help`, but they're defined in `web-mode`.

Even if we decided not to chance anything, there's an issue related to `describe-bindings`. It doesn't show the keymap defined in `help`. I couldn't figure this out. Any ideas?

**Comment by Ambrevar** [comment]
For the module question, actually I think the matter at hand here "what
should web-mode be"?

See #1851 for a discussion on web-mode, #1805 for the one about the
buffer class hierarchy.

At this point, I think web-mode has no reason to exist.

Instead, we should use mixins:

- history-buffer: enables history (web-buffers, internal-buffers, but
  NOT status-buffer, message-buffer).
- html-buffer: enables jump-to-headings, plump parsing, etc.

About the keymap: something fishy indeed.  I'll have a closer look next week.

**Comment by aadcg** [comment]
@Ambrevar, regarding the keybindings I was able to make sense of it with @aartaka. No need to look into it.

I'm making sense of our modules.

**Comment by aadcg** [comment]
We have 5 files sharing the same name both at `source/` and `source/mode`:

- bookmark
- help
- message
- no-procrastinate
- prompt-buffer

Shouldn't we at least append `mode` to those existing under `source/mode` to avoid confusion? 

Especially in our `defsystem` statement, I find it hard to follow.

**Comment by aadcg** [comment]
>     Shouldn't we at least append mode to those existing under
>     source/mode to avoid confusion?
>
>     Especially in our defsystem statement, I find it hard to follow.
>
> No, not necessarily. The mode/ directory is there exactly to mark those
> as modes. While you have to look at the :pathname in the defsystem, it's
> quite convenient otherwise.

It's exactly due to the :pathname detail that I bought it up.  I did
confuse me for a little while!

**Comment by aadcg** [comment]
I'll continue the work started here as soon as the 3.0 API will be closer to its final form. Thanks.

**Comment by aadcg** [comment]
Working on this again. 

The `headings-panel` command is broken. The problem was introduces in ef5a6f728f9c993a3a4d0f49d871440b7964ec0c. Investigating.
